06_gpu_and_ml/audio-to-text/parakeet.py

```diff
@@ -0,0 +1,246 @@
+# ---
+# cmd: ["modal", "serve", "06_gpu_and_ml/audio-to-text/parakeet.py::server_app"]
+# ---
+# # Real time audio transcription using Parakeet 🦜
+
+# [Parakeet](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/asr/models.html#parakeet) is the name of a family of ASR models built using [NVIDIA's NeMo Framework](https://docs.nvidia.com/nemo-framework/user-guide/latest/overview.html).
+# We'll show you how to use Parakeet for real-time audio transcription,
+# with a simple python client and a websocket server you can spin up easily in Modal.
+
+# This example uses the `nvidia/parakeet-tdt-0.6b-v2` model, which, as of May 13, 2025, sits at the
+# top of Hugging Face's [ASR leaderboard](https://huggingface.co/spaces/hf-audio/open_asr_leaderboard).
+
+# To run this example:
+
+# 1. Start the server locally:
+# ```bash
+# modal serve 06_gpu_and_ml/audio-to-text/parakeet.py::server_app
+# ```
+
+# 2. In a separate terminal, run the client:
+# ```bash
+# modal run 06_gpu_and_ml/audio-to-text/parakeet.py::client_app --modal-profile=$(modal profile current)
+# ```
+
+# See [Troubleshooting](https://modal.com/docs/examples/parakeet#client) at the bottom if you run into issues.
+
+# Here's what your final output might look like:
+
+# ```
+# ☀️ Waking up model, this may take a few seconds on cold start...
+
+# Recording and streaming... Press Ctrl+C to stop.
+# 📝 Transcription: Hi, how's it going?.
+# 📝 Transcription: Doing well. Great day to be having fun with transcription models
+# 📝 Transcription:
+# 📝 Transcription: Good.
+# ^C
+# 🛑 Stopped by user.
+# ```
+
+# ## Setup
+import modal
+
+app_name = "parakeet-websocket"
+
+server_app = modal.App(app_name)
+BUFFER_SIZE = 8000
+
+# ## Volume for caching model weights
+# We use a [Modal Volume](https://modal.com/docs/guide/volumes) to cache the model weights.
+# This allows us to avoid downloading the model weights every time we start a new instance.
+
+
+model_cache = modal.Volume.from_name("parakeet-model-cache", create_if_missing=True)
+# ## Configuring dependencies
+# The model runs remotely inside a [custom container](https://modal.com/docs/guide/custom-container). We can define the environment
+# and install our Python dependencies in that container's `Image`.
+
+# For inference, we recommend using the official NVIDIA CUDA Docker images from Docker Hub.
+# You'll need to add Python 3 and pip with the `add_python` option because the image
+# doesn't have these by default.
+
+# Additionally, we install `ffmpeg` for handling audio data and `fastapi` to create a web
+# server for our websocket. `uv` gives us even more speed on installation!
+
+image = (
+    modal.Image.from_registry(
+        "nvidia/cuda:12.8.0-cudnn-devel-ubuntu22.04", add_python="3.12"
```

Comment:
The version here is 12.8 and the cuda-python version differs -- previously 12.3, now 12.9.